src/services/cache_service.py

```
@@ -335,9 +335,26 @@ async def _cleanup_by_size(self) -> None:
+
+        try:
+            async with aiosqlite.connect(self.db_path) as db:
+                await db.execute("SELECT 1")
+            return True
+        except Exception as e:
+            logger.error(f"Cache service health check failed: {e}")
+            raise CacheError(
+                "Cache health check failed",
+                "health_check",
+                str(e),
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)